homeassistant/components/lastfm/sensor.py

```diff
@@ -35,95 +35,54 @@
 )
 
 
-def setup_platform(
+def format_track(track: Track) -> str:
+    """Format the track."""
+    return f"{track.artist} - {track.title}"
+
+
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Last.fm sensor platform."""
-    api_key = config[CONF_API_KEY]
-    users = config[CONF_USERS]
+    lastfm_api = LastFMNetwork(api_key=config[CONF_API_KEY])
+    async_add_entities(
+        (LastFmSensor(user, lastfm_api) for user in config[CONF_USERS]), True
+    )
 
-    lastfm_api = lastfm.LastFMNetwork(api_key=api_key)
 
-    entities = []
-    for username in users:
-        try:
-            lastfm_api.get_user(username).get_image()
-            entities.append(LastfmSensor(username, lastfm_api))
-        except WSError as error:
-            _LOGGER.error(error)
-            return
-
-    add_entities(entities, True)
-
-
-class LastfmSensor(SensorEntity):
+class LastFmSensor(SensorEntity):
     """A class for the Last.fm account."""
 
     _attr_attribution = "Data provided by Last.fm"
     _attr_icon = "mdi:radio-fm"
 
-    def __init__(self, user, lastfm_api):
+    def __init__(self, user: str, lastfm_api: LastFMNetwork) -> None:
         """Initialize the sensor."""
-        self._unique_id = hashlib.sha256(user.encode("utf-8")).hexdigest()
-        self._user = lastfm_api.get_user(user)
-        self._name = user
-        self._lastfm = lastfm_api
-        self._state = "Not Scrobbling"
-        self._playcount = None
-        self._lastplayed = None
-        self._topplayed = None
-        self._cover = None
-
-    @property
-    def unique_id(self):
-        """Return the unique ID of the sensor."""
-        return self._unique_id
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def native_value(self):
-        """Return the state of the sensor."""
-        return self._state
+        self._attr_unique_id = hashlib.sha256(user.encode("utf-8")).hexdigest()
+        self._attr_name = user
+        try:
+            self._user = lastfm_api.get_user(user)
+        except WSError as exc:
+            LOGGER.error(exc)
```

Comment:
You cannot do this inside `__init__`.
`__init__` method should not make any I/O.

Instead you should move this back to `setup_platform` and remove again the async migration.